grakn-core/src/main/java/ai/grakn/exception/GraknBackendException.java

```diff
@@ -18,19 +18,170 @@
 
 package ai.grakn.exception;
 
+import ai.grakn.engine.TaskId;
+import ai.grakn.graql.Query;
 import ai.grakn.util.ErrorMessage;
+import org.apache.commons.lang.exception.ExceptionUtils;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static ai.grakn.util.ErrorMessage.BACKEND_EXCEPTION;
+import static ai.grakn.util.ErrorMessage.STATE_STORAGE_ERROR;
 
 /**
- * An exception which encapsulates a vendor backend error
+ * <p>
+ *     Backend Grakn Exception
+ * </p>
+ *
+ * <p>
+ *     Failures which occur server side are wrapped in this exception. This can include but is not limited to:
+ *     - Kafka timeouts
+ *     - Cassandra Timeouts
+ *     - Malformed Requests
+ * </p>
  *
- * @author Filipe Teixeira
+ * @author fppt
  */
-public class GraknBackendException extends GraphRuntimeException {
-    public GraknBackendException(Exception e) {
-        super(ErrorMessage.BACKEND_EXCEPTION.getMessage(), e);
+public class GraknBackendException extends GraknException {
+    private Optional<Integer> status = Optional.empty();
```

Comment:
I am not happy with this approach either and was trying to avoid declaring a sub-class but I suppose there is a good enough cause for it in this case.